cgrep.py

```diff
@@ -0,0 +1,443 @@
+    print("You can only use -i with -t or by itself")
+
+  # if -i and -t
+  elif options.token and options.ip:
+    try:
+      get_nets([options.token], db)
+    except naming.UndefinedAddressError as e:
+      print("Network group '%s' is not defined!" % (e.message.split()[0]))
+    else:
+      results = compare_ip_token(options, db)
+      print(results)
+
+  # if -t, but not -i; invalid!
+  elif options.token and not options.ip:
+    print("You must specify an IP Address with -i [addr]")
+
+  # if -i
+  elif options.ip:
+    for ip in options.ip:
+      groups = get_ip_parents(ip, db)
+      print("Results for IP: %s" % ip)
+      # iterate and print the tokens we found.
+      for name, networks in groups:
+        # print the group name [0], and the networks it was in [1]
+        print("%s  %s" % (name, networks))
+
+  elif options.gmp:
+    common, diff1, diff2 = group_diff(options, db)
+    print_diff(options.gmp[0], common, diff1, diff2)
+    print('')
+    print_diff(options.gmp[1], common, diff2, diff1)
+
+  # if -c
+  elif options.cmp:
+    meta, results = compare_tokens(options, db)
+    print('Union of %s and %s:\n %s\n' % meta)
+    print('Diff of %s and %s:' % meta[:-1])
+    for i in results:
+      print('  ' + i)
+    print('')
+    first_obj, sec_obj = options.cmp
+    if check_encapsulated("network", first_obj, sec_obj, db):
+      print("%s fully encapsulates %s" % (sec_obj, first_obj))
+    else:
+      print("%s does _not_ fully encapsulate %s" % (sec_obj, first_obj))
+    # check the other way around.
+    if check_encapsulated("network", sec_obj, first_obj, db):
+      print("%s fully encapsulates %s" % (first_obj, sec_obj))
+    else:
+      print("%s does _not_ fully encapsulate %s" % (first_obj, sec_obj))
+
+  # if -o
+  elif options.obj:
+    for obj in options.obj:
+      try:
+        token, ips = get_nets([obj], db)[0]
+      except naming.UndefinedAddressError:
+        print("%s is an invalid object" % obj)
+      else:
+        print(token + ":")
+        # convert list of ip objects to strings and sort them
+        ips.sort(key=lambda x: int(x.ip))
+        p([str(x) for x in ips])
+
+  # if -s
+  elif options.svc:
+    try:
+      results = get_ports(options.svc, db)
+    except naming.UndefinedServiceError:
+      print("%s contains an invalid service object" % str(options.svc))
+    else:
+      for result in get_ports(options.svc, db):
+        svc, port = result
+        print(svc + ":")
+        p(port)
+
+  # if -p
+  elif options.port:
+    port, protocol, result = get_services(options, db)
+    print("%s/%s:" % (port, protocol))
+    p(result)
+
+  # if nothing is passed
+  elif not any((options.cmp, options.ip, options.token, options.obj,
+               options.svc, options.port)):
+    parser.print_help()
+  print('')
+
+
+def check_encapsulated(obj_type, first_obj, second_obj, db):
+  """ Checks if a network/service object is entirely contained within another.
```

Comment:
I see this method raises an error, usually this is documented in the doc strings also.